### PR TITLE
How-To Guides section

### PR DESCRIPTION
**Pull request by taylorsilva**
Pre-populating with some of the blog posts I made.

### CODE REVIEW

**Comment by chenbh** [comment]
@clarafu

File: lit/docs/guides/pipeline-guides.lit
Comment (clarafu) [inline lit/docs/guides/pipeline-guides.lit:107]:
```suggestion

    Using the task in a pipeline.
```

I was thinking an extra line break before that line might look nice here, what do you think?

Screen Shot 2021-05-18 at 3 01 50 PM

File: lit/docs/guides/pipeline-guides.lit
Comment (clarafu) [inline lit/docs/guides/pipeline-guides.lit:143]: Weird it seems to remove the curly braces from the docs? :( Is it just my build that doesn't show it?

Screen Shot 2021-05-18 at 3 09 50 PM

Maybe we should just move the `repository: busybox` onto a new line instead, so that it doesn't look like you can put it on the same line without the curly braces!

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:82]:
```suggestion
    plan:
```

Just added an extra space to align!

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:83]:
```suggestion
    - get: concourse-examples
```

Just added an extra space to align here too!

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:104]: Ah looks like all the job configs at the `plan:` and step lines are one space unaligned in this file haha

Screen Shot 2021-05-18 at 3 25 45 PM

File: lit/docs/guides/git.lit
Comment (clarafu) [inline lit/docs/guides/git.lit:7]: Are you working on the `Git Guides` content within this PR? Or if we want to write the content for it in a later PR, we could remove this section for now and then add it back later?

**Review by clarafu — COMMENTED** [review]
Really great content! 😄  I think this would be super useful to have within our docs because it talks about different features within Concourse that beginner users would not know about unless they dug really deep into our docs. Also its nice to have more docs about how to write Concourse pipelines and different real-world scenarios for them with clear explanations. 

Just had a few comments on mostly formatting, but the content looks great to me!

File: lit/docs/guides/git.lit
Comment (taylorsilva) [reply lit/docs/guides/git.lit:7]: I have a one that I want to add as part of this PR. Wanted to submit what I had so far to get feedback on the direction 😀

File: lit/docs/guides/pipeline-guides.lit
Comment (taylorsilva) [reply lit/docs/guides/pipeline-guides.lit:143]: yeah that's a weird thing, gotta do TRIPLE CURLY BRACES when declaring the codeblock to avoid this. Gonna put it on a newline instead.

**Comment by taylorsilva** [comment]
@clarafu done and ready for final review! I won't be adding any more guides within this PR

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:153]: nit: This is one space unaligned! Not sure if you want to bother fixing all of them, theres a few more places where it is one space off haha

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:184]: Like here too!

File: lit/docs/guides/container-images.lit
Comment (clarafu) [inline lit/docs/guides/container-images.lit:218]: Annnnd here 😅

**Review by clarafu — COMMENTED** [review]
minor nit comments, but it LGTM!

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:64]: minor thing, but it's a bit awkward to suggest that you typically want to wrap your `get` steps in an `in_parallel`, but not follow that advice in later examples

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:94]:
```suggestion
    \title{Putting Task Configs in Files}
```

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:122]:
```suggestion
    \titled-codeblock{pipeline.yml}{yaml}{{
```

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:184]: I feel like this would be less confusing to newcomers if we left the `image_resource` out in this example. We're introducing it as a one-or-the-other thing, and the fact that it takes precedence is maybe better left to just the docs.

Speaking of, maybe we can link to the relevant places in the docs for `image_resource` and `task.image` here too?

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:143]: Maybe we could touch on *why* you might want to do this (for instance, you want to trigger a job on new versions of an image)

File: lit/docs/guides/pipeline-guides.lit
Comment (aoldershaw) [inline lit/docs/guides/pipeline-guides.lit:220]: the `{...}` get clobbered here:

Screen Shot 2021-05-21 at 12 37 25 PM

File: lit/docs/guides/git.lit
Comment (aoldershaw) [inline lit/docs/guides/git.lit:50]: Perhaps this is a little misleading, as the image used in this task doesn't need `git`

File: lit/docs/guides/git.lit
Comment (aoldershaw) [inline lit/docs/guides/git.lit:100]:
```suggestion
            - |
              cd repo-main
              # edit a file / make a change
              date +%Y-%m-%d > todays-date
              git add ./todays-date
              git commit -m "Add todays date"
              git tag v0.1.6
```

if the text is less indented than the `|`, it's invalid YAML

**Review by aoldershaw — COMMENTED** [review]
A couple small suggestions. Looks great though!

File: lit/docs/guides/pipeline-guides.lit
Comment (taylorsilva) [reply lit/docs/guides/pipeline-guides.lit:64]: Had this thought too as I was writing it 😅 I ended up not adding `in_parallel` because I wanted to keep the example focused on the put step farther down. That all said, I think I'm overthinking it, might as well follow recommended practices everywhere!

File: lit/docs/guides/pipeline-guides.lit
Comment (taylorsilva) [reply lit/docs/guides/pipeline-guides.lit:184]: Really great suggestion. Changed the first paragraph to link to both.

**Comment by taylorsilva** [comment]
@clarafu @aoldershaw thank you both for looking this over and finding all those nit's. It's amazing how many times these pages an be read and there are STILL little mistakes all over 😅

**Review by clarafu — APPROVED** [review]
It LGTM! :)

**Comment by taylorsilva** [comment]
@vito can't merge until someone else on the docs team approves

**Review by vito — APPROVED** [review]
Approving!

**Pull request merged**
